test/apps/iodemo/io_demo.cc

```diff
@@ -1292,7 +1301,11 @@ class DemoServer : public P2pDemoCommon {
 
         VERBOSE_LOG << "got io (AM) message " << io_op_names[msg->op] << " sn "
                     << msg->sn << " data size " << msg->data_size
-                    << " conn " << conn;
+                    << " conn " << conn;  
+
+#ifndef NDEBUG
+        conn->set_remote_id(msg->conn_id);
```

Comment:
i do not think it makes much sense, here you allways set `remote_id` to what is received in `msg`,
then you check in `handle_io_am_read_request` and `handle_io_am_read_request`
this check will always be true

Maybe setting the id is enough because you have an assert inside?

Reply:
@brminich do you mean check at lines 926-930? yes, with latest change it seems to be a double check

Reply:
then is the previous solution ok for you? setting only when receiving the first message

Reply:
IMO better to keep code simpler, I would keep unconditional initialization with assertion inside and del 926-930